scrapy/utils/defer.py

```diff
@@ -145,7 +130,7 @@ def parallel(
     return DeferredList([coop.coiterate(work) for _ in range(count)])
 
 
-class _AsyncCooperatorAdapter(Iterator[Deferred], Generic[_T]):
+class _AsyncCooperatorAdapter(Iterator[Deferred], Generic[_T, _SelfResultT]):
```

Comment:
This causes test failures, why is this needed?

Reply:
See full error stack trace here: https://github.com/vsevolodbreus/scrapy/actions/runs/10700184494/job/29663370539

Reply:
Looks like it was fixed in some 3.10.x (it works on 3.10.14 and doesn't work on 3.10.5, maybe it's related to "gh-98852: Fix subscription of types.GenericAlias instances containing bare generic types: for example tuple[A, T][int], where A is a generic type, and T is a type variable." from 3.10.9) and 3.11+ and wasn't fixed in 3.9. And it looks like `typing.Iterator` works here so there is a workaround.